src/app/utils.ts

```diff
@@ -540,3 +540,66 @@ export function getLocalStoragePaginationValue(id?: string) {
 
   return undefined;
 }
+
+export const searchPairs = (query: string, pairsList: any): any => {
+  const searchQuery = query.trim().toLowerCase().replace(/\s+/g, " ");
+
+  const hasTypoTolerance = (source: string, target: string): boolean => {
```

Comment:
The sliding window approach to check for typos is straight forward and takes into account character mismatches, so its a great start and we can keep it. But for future reference, can we add a TODO here to switch to use the Levenstein distance, which also takes into account missing characters (deletion) or extra characters (insertion), and is a more common approach to handle typo tolerance. 

```ts
// TODO: Implement typo tolerance using Levenshtein distance instead of sliding window approach.
```